packages/core/src/CSL/cslToCore/certificate.ts

```diff
@@ -34,52 +35,54 @@ const stakeDelegation = (certificate: CSL.StakeDelegation): StakeDelegationCerti
   stakeKeyHash: Ed25519KeyHash(Buffer.from(certificate.stake_credential().to_keyhash()!.to_bytes()).toString('hex'))
 });
 
-const createCardanoRelays = (relays: CSL.Relays): Relay[] => {
-  const result: Relay[] = [];
-  for (let i = 0; i < relays.len(); i++) {
-    const relay = relays.get(i);
-    const relayByAddress = relay.as_single_host_addr();
-    const relayByName = relay.as_single_host_name();
-    const relayByNameMultihost = relay.as_multi_host_name();
+const createCardanoRelays = (relays: CSL.Relays): Relay[] =>
+  usingAutoFree((scope) => {
+    const result: Relay[] = [];
+    for (let i = 0; i < relays.len(); i++) {
+      const relay = scope.manage(relays.get(i));
+      const relayByAddress = relay.as_single_host_addr();
+      const relayByName = relay.as_single_host_name();
+      const relayByNameMultihost = relay.as_multi_host_name();
 
-    if (relayByAddress) {
-      // RelayByAddress
-      result.push({
-        __typename: 'RelayByAddress',
-        ipv4: relayByAddress.ipv4()?.ip().join('.'),
-        ipv6: relayByAddress.ipv6()?.ip().join('.'),
-        port: relayByAddress.port()
-      });
+      if (relayByAddress) {
+        // RelayByAddress
+        result.push({
+          __typename: 'RelayByAddress',
+          ipv4: relayByAddress.ipv4()?.ip().join('.'),
+          ipv6: relayByAddress.ipv6()?.ip().join('.'),
+          port: relayByAddress.port()
```

Comment:
Those probably needs to be managed

Reply:
214d62f11644c689a1e38c3f1811f88e87acf8ac